src/seedu/addressbook/commands/Command.java

```diff
@@ -70,4 +70,11 @@ public int getTargetIndex() {
     public void setTargetIndex(int targetIndex) {
         this.targetIndex = targetIndex;
     }
+
+    /**
+     * Returns true if the command will mutate data in the address book.
+     */
+    public boolean isMutating(){
+        throw new UnsupportedOperationException("This method should be implemented in child classes");
```

Comment:
A better way is to define this as an abstract class. This will force implementation by child classes and you need not throw an exception.